core/base/combination.cpp

```diff
@@ -60,6 +60,67 @@ inline void initialize_scalars(std::shared_ptr<const Executor> exec,
 }  // namespace
 
 
+template <typename ValueType>
+Combination<ValueType>& Combination<ValueType>::operator=(
+    const Combination& other)
```

Comment:
yes, and another issue there is that the type implementing operator= needs to know the executor, because otherwise you are able to assign a CUDA system matrix to a OpenMP solver that was initialized as an empty object (create_default()).